modules/tensor_mechanics/include/materials/HillConstants.h

```diff
@@ -58,4 +73,8 @@ class HillConstants : public ADMaterial
 
   /// The functions describing the temperature dependence
   std::vector<const Function *> _functions;
+
+private:
+  /// Compute Euler angles in <Z Y X> sequence from strain kinematic rotation matrix
+  std::array<Real, 3> computeZYXAngles(const RankTwoTensor & rotation_matrix);
```

Comment:
We have an `EulerAngles` class that this function should be added to. And if I'm not mistaken @SudiptaBiswas did implement a rotation matrix -> euler angles routine before.

Reply:
Here: https://github.com/idaholab/moose/blob/e65c36f92f31449296c2cb86a1cd463e0cb3c65f/modules/phase_field/src/userobjects/EulerAngleUpdater.C#L108-L155

Reply:
Looks like the angle sequence there is different.

Reply:
Yes, but does it have to be? It would be less confusing if we had one single convention for Euler angles in MOOSE.

Reply:
`EulerEngleUpdater` follows ZX'Z'' transformation and it was designed for a specific application. I agree with @dschwen, we should refactor this angle extraction operation in a separate class.

Reply:
> Yes, but does it have to be?

With the existing code, yes, it has to because of how the transformation of the Hill tensor was done. We would need to also change that to have the same sequence as the one present in the phase field module.

Reply:
The modifications last pushed transform the Hill's tensor in Voigt notation using the components of the total rotation matrix, so there isn't a need to define rotation sequences. For all the anisotropic classes, the transformed matrix will be used by default. An additional option is provided to the user when using the creep class: The Hill coefficients are extracted at the beginning of the solution (`use_large_rotation = false`) and the ADHillCreep uses such coefficients by prescribing `use_transformation = false`. This is done to avoid doing unnecessary computations when transformations aren't needed, but, from the tests, the performance differences may not be very large.